### PR TITLE
fuzzing: print the diff in case of issue

### PR DESCRIPTION
**Pull request by sylvestre**

### CODE REVIEW

File: fuzz/Cargo.toml [hunk 1]
```diff
@@ -12,6 +12,7 @@ libfuzzer-sys = "0.4"
 libc = "0.2"
 tempfile = "3"
 rand = { version = "0.8", features = ["small_rng"] }
+similar = "1.3.0"
```
Comment (cakebaker) [inline fuzz/Cargo.toml:15]: Why don't you use the latest version `2.4.0`?

File: fuzz/fuzz_targets/fuzz_common.rs [hunk 2]
```diff
@@ -353,6 +357,21 @@ pub fn compare_result(
...
+/// When we have different outputs, print the diff
+fn print_diff(rust_output: &str, gnu_output: &str) {
+    println!("Diff=");
+    let diff = TextDiff::from_lines(rust_output, gnu_output);
+    for change in diff.iter_all_changes() {
+        let sign = match change.tag() {
+            ChangeTag::Delete => "-",
+            ChangeTag::Insert => "+",
+            ChangeTag::Equal => " ",
+        };
```
Comment (cakebaker) [inline fuzz/fuzz_targets/fuzz_common.rs:369]: This seems to be unnecessary because `ChangeTag` implements `Display`.

Comment (sylvestre) [reply fuzz/fuzz_targets/fuzz_common.rs:369]: are you sure ? I can't make it work 

```
fn print_diff(rust_output: &str, gnu_output: &str) {
    let diff = TextDiff::from_lines(rust_output, gnu_output);
    println!("Diff=");
    for change in diff.iter_all_changes() {
        print!("{}", change);
    }
    println!();
}

```

Comment (cakebaker) [reply fuzz/fuzz_targets/fuzz_common.rs:369]: Yes :)

`ChangeTag` is only about the sign. So you would have to use:
```
for change in diff.iter_all_changes() {
    print!("{}{}", change.tag(), change);
}
```

Comment (sylvestre) [reply fuzz/fuzz_targets/fuzz_common.rs]: TIL, thanks :)

**Comment by sylvestre** [comment]
great output here:
https://github.com/uutils/coreutils/issues/5849

**Pull request merged**